src/Schema.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace BuildEmpire\Apartment;
+
+use BuildEmpire\Apartment\Exceptions\NoSchemaSetException;
+use BuildEmpire\Apartment\Exceptions\NoSchemaFoundException;
+use BuildEmpire\Apartment\Exceptions\SchemaNameNotValidException;
+use BuildEmpire\Apartment\Helpers\ApartmentHelpers;
+
+class Schema
+{
+    protected $schemaName = false;
+
+    /**
+     * Set the schema name.
+     *
+     * @param $schemaName
+     * @throws SchemaNameNotValidException
+     */
+    public function setSchemaName($schemaName)
+    {
+        if (!ApartmentHelpers::isSchemaNameValid($schemaName)) {
+            throw new SchemaNameNotValidException('The apartment ' . $schemaName . ' is not valid. It must be all lowercase and only contain letters, numbers, or underscores.');
+        }
+        $this->schemaName = $schemaName;
+    }
+
+    /**
+     * Try to get the schema name.
+     *
+     * @return string
+     * @throws NoSchemaSetException
+     */
+    public function getSchemaName()
+    {
+        if ($this->schemaName !== false) {
+            return $this->schemaName;
+        }
+
+        throw new NoSchemaSetException('No schema is set. If you want to check if a schema exists use doesSchemaExist method.');
+    }
+
+    /**
+     * Is the schema currently set?
+     *
+     * @return bool
+     */
+    public function isSchemaSet()
+    {
+        return (!$this->schemaName === false);
```

Comment:
Is this the intended logic here (effectively `(!$this->schemaName) === false`, the parentheses in the original code wrap the lowest precedence operator and so don't do anything)?  In which case, isn't this just `return (boolean)$this->schemaName;` (I believe that's the php equivalent of `return !!some_non_boolean` in js)?